File: chain-sdk/clojure/src/on_chain_api/specs/token_dto.clj
(ns on-chain-api.specs.token-dto
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def token-dto-data
  {
   (ds/opt :entry_time) inst?
   (ds/opt :recv_time) inst?
   (ds/opt :block_number) int?
   (ds/opt :id) string?
   (ds/opt :factory) string?
   (ds/opt :symbol) string?
   (ds/opt :name) string?
   (ds/opt :decimals) string?
   (ds/opt :total_supply) string?
   (ds/opt :volume) string?
   (ds/opt :volume_usd) string?
   (ds/opt :untracked_volume_usd) string?
   (ds/opt :tx_count) string?
   (ds/opt :liquidity) string?
   (ds/opt :derived_eth) string?
   (ds/opt :whitelist_pairs) (s/coll-of string?)
   (ds/opt :vid) int?
   (ds/opt :token_symbol) string?
   })

(def token-dto-spec
  (ds/spec
    {:name ::token-dto
     :spec token-dto-data}))

File: chain-sdk/javascript-closure-angular/API/Client/TokenDTO.js
goog.provide('API.Client.TokenDTO');

/**
 * Stores aggregated information for a specific token across all pairs that token is included in.
 * @record
 */
API.Client.TokenDTO = function() {}

/**
 * @type {!Date}
 * @export
 */
API.Client.TokenDTO.prototype.entryTime;

/**
 * @type {!Date}
 * @export
 */
API.Client.TokenDTO.prototype.recvTime;

/**
 * Number of block in which entity was recorded.
 * @type {!number}
 * @export
 */
API.Client.TokenDTO.prototype.blockNumber;

/**
 * Token address.
 * @type {!string}
 * @export
 */
API.Client.TokenDTO.prototype.id;

/**
 * Factory address.
 * @type {!string}
 * @export
 */
API.Client.TokenDTO.prototype.factory;

/**
 * Token symbol.
 * @type {!string}
 * @export
 */
API.Client.TokenDTO.prototype.symbol;

/**
 * Token name.
 * @type {!string}
 * @export
 */
API.Client.TokenDTO.prototype.name;

/**
 * Token decimals.
 * @type {!string}
 * @export
 */
API.Client.TokenDTO.prototype.decimals;

/**
 * Total supply of liquidity token.
 * @type {!string}
 * @export
 */
API.Client.TokenDTO.prototype.totalSupply;

/**
 * Amount of token traded all time across all pairs.
 * @type {!string}
 * @export
 */
API.Client.TokenDTO.prototype.volume;

/**
 * Amount of token in USD traded all time across pairs (only for tokens with liquidity above minimum threshold).
 * @type {!string}
 * @export
 */
API.Client.TokenDTO.prototype.volumeUsd;

/**
 * Amount of token in USD traded all time across pairs (no minimum liquidity threshold).
 * @type {!string}
 * @export
 */
API.Client.TokenDTO.prototype.untrackedVolumeUsd;

/**
 * Amount of transactions all time in pairs including token.
 * @type {!string}
 * @export
 */
API.Client.TokenDTO.prototype.txCount;

/**
 * Total amount of token provided as liquidity across all pairs.
 * @type {!string}
 * @export
 */
API.Client.TokenDTO.prototype.liquidity;

/**
 * ETH per token.
 * @type {!string}
 * @export
 */
API.Client.TokenDTO.prototype.derivedEth;

/**
 * Array of whitelisted pairs.
 * @type {!Array<!string>}
 * @export
 */
API.Client.TokenDTO.prototype.whitelistPairs;

/**
 * 
 * @type {!number}
 * @export
 */
API.Client.TokenDTO.prototype.vid;

/**
 * @type {!string}
 * @export
 */
API.Client.TokenDTO.prototype.tokenSymbol;

